### PR TITLE
refactor: cpda rename instruction to invoke

### PR DESCRIPTION
**Pull request by ananas-block**

### CODE REVIEW

File: examples/token-escrow/programs/token-escrow/tests/test_compressed_pda.rs [hunk 1]
```diff
@@ -14,8 +14,10 @@
 // release compressed tokens
```
Comment (SwenSchaeferjohann) [inline examples/token-escrow/programs/token-escrow/tests/test_compressed_pda.rs:14]: is this a todo?

File: programs/compressed-pda/src/invoke_cpi/process_cpi_context.rs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+pub fn set_cpi_context(
+    cpi_context_account: &mut CpiContextAccount,
+    inputs: InstructionDataInvokeCpi,
+) {
+    // Check conditions and modify the signatures
+    if cpi_context_account.context.is_empty() {
+        msg!("cpi signatures are empty");
+        // cpi signature account should only be used with mutiple compressed accounts owned by different programs
+        // thus the first invocation execute is assumed to be false
```
Comment (SwenSchaeferjohann) [inline programs/compressed-pda/src/invoke_cpi/process_cpi_context.rs:41]: is this meant to say `invocation executed`?

File: programs/compressed-pda/src/sdk/invoke.rs [hunk 3]
```diff
@@ -142,31 +141,27 @@ pub fn create_execute_compressed_instruction(
...
-    let instruction_data = crate::instruction::ExecuteCompressedTransaction { inputs };
+    let instruction_data = crate::instruction::Invoke { inputs };
 
     let compressed_sol_pda = compression_lamports.map(|_| get_compressed_sol_pda());
 
-    let accounts = crate::accounts::TransferInstruction {
+    let accounts = crate::accounts::InvokeInstruction {
         fee_payer: *fee_payer,
         authority: *payer,
         // authority_pda: get_cpi_authority_pda(&crate::ID),
```
Comment (SwenSchaeferjohann) [inline programs/compressed-pda/src/sdk/invoke.rs:157]: leftover?

File: programs/compressed-pda/src/sdk/invoke.rs [hunk 4]
```diff
@@ -219,13 +214,13 @@ pub fn create_execute_compressed_instruction(
 
 //     let mut inputs = Vec::new();
 
-//     let accounts = crate::accounts::TransferInstruction {
+//     let accounts = crate::accounts::InvokeInstruction {
```
Comment (SwenSchaeferjohann) [inline programs/compressed-pda/src/sdk/invoke.rs:217]: can these be removed?

**Review by SwenSchaeferjohann — APPROVED** [review]
nice!

just left some questions on commented code

File: examples/token-escrow/programs/token-escrow/tests/test_compressed_pda.rs (hunk 1 above)
Comment (ananas-block) [reply examples/token-escrow/programs/token-escrow/tests/test_compressed_pda.rs:14]: yes

File: programs/compressed-pda/src/invoke_cpi/process_cpi_context.rs (hunk 2 above)
Comment (ananas-block) [reply programs/compressed-pda/src/invoke_cpi/process_cpi_context.rs:41]: outdated comment, the CompressedCpiContext object hat a property ``execute`` before. I replaced it with ``set_context``

File: programs/compressed-pda/src/sdk/invoke.rs (hunk 4 above)
Comment (ananas-block) [reply programs/compressed-pda/src/sdk/invoke.rs:217]: let's remove this when we merge the optimized compressed pda instruction data

**Pull request merged**